src/sulogin.c

```diff
@@ -133,8 +131,8 @@ main(int argc, char **argv)
 	(void) signal (SIGALRM, catch_signals);	/* exit if the timer expires */
 	(void) alarm (ALARM);		/* only wait so long ... */
 
-	while (true) {		/* repeatedly get login/password pairs */
-		char *cp;
+	while (!done) {		/* repeatedly get login/password pairs */
```

Comment:
This would read better as a do-while loop, and that would itself allow removing the initialization of `bool done = false;`

Reply:
Also, you could maybe refactor to a do-while loop in a separate commit before the one that removes the `pass` array.

Reply:
I have applied both suggestions.